lib/faraday/http_cache/response.rb

```diff
@@ -192,10 +196,10 @@ def cache_control
       #
       # Returns nothing.
       def wrap_headers!
-        headers = @payload[:response_headers]
+        headers_hash = @payload[:response_headers]
```

Comment:
If the `header` name wasn't conflicting with anything else I rather leave it as is and only change what's necessary to fix the bug regarding the `Date` header.